### DISCUSSION TITLE
FastAPI integration with Okta using OpenIDConnect

### DISCUSSION THREAD

**Discussion by gopalsi** [fastapi/fastapi#9101] [Questions]
I am trying to integrate FastAPI authentication with Okta using OpenIDConnect.
I could find library for Flask https://flask-oidc.readthedocs.io/en/latest/ but could not find something similar for FastAPI.

Can someone please help with the same.

**Comment by Kludex** [comment]
I've found this: https://github.com/tiangolo/fastapi/issues/488
Could you let us know if this is somehow helpful?

**Comment by gopalsi** [comment]
Thanks for sharing #488 , I went through it, but it doesn't seems to be of much help as Okta authentication works differently.

**Comment by GH-maggio** [comment]
@gopalsi authlib has a fastapi intergration that might work: https://docs.authlib.org/en/latest/client/fastapi.html

**Comment by pavelanni** [comment]
@gopalsi Did you find the solution for Okta + FastAPI integration? I am looking for something like this Flask integration: https://developer.okta.com/blog/2018/07/12/flask-tutorial-simple-user-registration-and-login or this: https://github.com/okta/samples-python-flask . I am sure it's doable, but I couldn't find any examples.

**Comment by HarryMWinters** [comment]
FWIW I implemented a solution [here](https://github.com/HarryMWinters/fastapi-oidc).
